### ISSUE TITLE
Quicksilver fails on simple task

### ISSUE THREAD

**Issue by gmcinnes** [pantheon-systems/quicksilver-examples#61]
I"m adding this issue here on the advice of Zvi Epner at Pantheon support.

I have the following simple test script:

```
<?php
print 'QS test script';
passthru("wp search-replace https://www.marsdd.com http://db-clean-www-marsdd-com.pantheonsite.io —verbose 2>&1");
```

* It does not appear to run
* No output appears from it in the workflow log (even the first print statement)
* No failure states are mentioned in the workflow log
* When I remove the `passthru` statement the print output shows up in the workflow log

`wp search-replace` is quite a long running process.  I suspect it is being killed by a timeout, or other resource limitation.  A mention of it does happen in php slow log.  Zvi was unable to confirm or deny this though, and I don't see any documentation of resource limitations for quicksilver anywhere.

It might  be nothing to do with that though.  It would be great if someone could look into this and give me a hint as to what might be going wrong.

**Comment by joshkoenig** [comment]
Currently quicksilver operations are limited to a 120 second timeout. If you're not getting results, that's likely the reason why. I think it's far to say we should make that clear, which I'll check up on.

More generally, you should be able to trigger this same operation (with no timeouts) as a standard part of the database clone workflow within the UI.

**Comment by gmcinnes** [comment]
Hi @joshkoenig .   I'm going to enumerate the issues for brevity here.  Don't read anything else into my terseness.

WRT the use case for this:

* I know I *should* be able to trigger this using the UI, but for some reason it doesn't work on all my sites
* One site in particular has issues with the urls of the main navigation not being replaced, which makes it a huge pain to remember fix with `terminus wp "search-replace foo bar" every time I make a multidev environment
* In particular I have a ticket open with pantheon support since June 2015 about this that wasn't really resolved.  I didn't push the issue.
* Also, it would be handy to *force* this to happen automatically every time the database is cloned from live.

As for the possible timeout:

* My db is bigger than average (174MB in the backups).  It *does* take a bit of time to do a `wp search-replace` locally, but the db is mostly full of audit logs that my client doesn't want to lose, so I can't do much about it.
* Can someone confirm that's the issue?
* If so, I agree, you should document it
* There should be some sort of output in the workflow logs in this failure case.  As I noted there's not only no record of the workflow failing, there's no record of it even starting.
* As well, any debugging statements, even ones added *before* the potentially problematic, long running code, also don't make it to the workflow log, which is certainly violating the PLOS.

I'm not sure this is the right place to discuss this, 'cause it doesn't really seem like an issue related to a  "quicksilver-example".  Let me know if you'd prefer to move this off thread.

**Comment by joshkoenig** [comment]
> I know I should be able to trigger this using the UI, but for some reason it doesn't work on all my sites.

By any chance do the sites where this doesn't work have no WordPress icon on your home screen? If they aren't properly associated with WP as their "framework" then the WP-specific dashboard options will not be present. 

> It does take a bit of time to do a wp search-replace locally.

If it takes over 120 seconds, this is very likely the issue. This is documented, but it's easy to miss. I'll follow up with our support team to dig in and make sure this is the case. Clearly room for improvement here.

**Comment by gmcinnes** [comment]
> By any chance do the sites where this doesn't work have no WordPress icon on your home screen? If they aren't properly associated with WP as their "framework" then the WP-specific dashboard options will not be present.

No. The options are there.  They just don't work.  Some urls are replaced, some are not.  When I run a search-replace with terminus it replaces the rest of them and fixes it.

> If it takes over 120 seconds, this is very likely the issue. This is documented, but it's easy to miss. I'll follow up with our support team to dig in and make sure this is the case. Clearly room for improvement here.

FWIW, on my local dev machine that looks like:

```
time wp search-replace https://www.marsdd.com http://marsdd.dev:8888

.....


Success: Made 113123 replacements.

real	1m33.815s
user	0m0.817s
sys	0m0.126s
```

so, less than 120 seconds, but I have no way of knowing how long it would take inside a Pantheon container.

**Comment by joshkoenig** [comment]
> No. The options are there. They just don't work. Some urls are replaced, some are not. When I run a search-replace with terminus it replaces the rest of them and fixes it.

Well that's weird. They do the exact same thing on the back-end. I'll look up the case number.

> so, less than 120 seconds, but I have no way of knowing how long it would take inside a Pantheon container.

Likely a bit longer if only because of the remote DB connection. That adds a few milisecond penalty to each request vs when it's all on one machine. if you're doing 100k replacements, that adds up. We've [sponsored some improvements](https://pantheon.io/blog/search-and-replace-love-and-profit) to the performance of WP-CLI that should help speed things up in general, but they're only rolling out now.

We've identified the need to get better feedback on timeouts and should be able to improve this situation in the coming weeks. I'll follow up on your support tickets with specific confirmation that this was the cause, as well as looking into why you're not able to use the time-unlimited built-in dashboard function to take care of your search-and-replacing.

**Comment by joshkoenig** [comment]
This is now partially fixed. We are handling timeouts properly internally. 

Still work to do in order to get the timeouts to be presented nicely in the Terminus `workflows --watch` output.
